Include all parameters in the demo app's layout test window

justincredible opened the pull request:
<!--
Please read the "Making a PR" section of [`CONTRIBUTING.md`](https://github.com/emilk/egui/blob/main/CONTRIBUTING.md) before opening a Pull Request!

* Keep your PR:s small and focused.
* The PR title is what ends up in the changelog, so make it descriptive!
* If applicable, add a screenshot or gif.
* If it is a non-trivial addition, consider adding a demo for it to `egui_demo_lib`, or a new example.
* Do NOT open PR:s from your `master` branch, as that makes it hard for maintainers to test and add commits to your PR.
* Remember to run `cargo fmt` and `cargo clippy`.
* Open the PR as a draft until you have self-reviewed it and run `./scripts/check.sh`.
* When you have addressed a PR comment, mark it as resolved.

Please be patient! I will review your PR, but my time is limited!
-->

* [x] I have followed the instructions in the PR template

This change adds the `main_align` and `main_justify` options to the "Layout Test" demo window.
<img width="1004" height="609" alt="demo_layout_test" src="https://github.com/user-attachments/assets/89824e09-c559-4009-b5fd-3d6a8bcae3ca" />

Since all `Layout` fields are being used I replaced the `LayoutSettings` struct, but it required uncommenting the serde cfg_attr derives. This was commented out in bf19eb2ec5977a389538cff23b2041b27ec10011 (and the feature name subsequently kept up-to-date), but it's not clear to me why so I can revert that commit and just update the demo struct if necessary.

The button's size was increased to demonstrate the text alignment, and the `Resize::max_size()` works around auto-resizing for `main_justify`.

justincredible committed:
Replace `LayoutSettings` with `Layout`

justincredible committed:
Add missing layout parameters in demo

The resize region needs to be limited or else it expands beyond the
viewport boundaries when main justify is set. The button needs to be
larger than default to see the how alignment affects the components' text.

justincredible committed:
Add missing layout parameters

The resize region needs to be limited or else it expands beyond the
viewport boundaries when main justify is set. The button needs to be
larger than default to see the how alignment affects the components' text.

justincredible committed:
Update snapshot

justincredible committed:
Update snapshot

justincredible committed:
Update doc comments for layout fields

justincredible committed:
Update doc comments for layout fields

lucasmerlin committed:
Add reasonable timeouts to all workflows (#7499)

Turns out the default timeout for github actions is 6 hours (!). This PR
sets some reasonable default for all workflows, the ones invoking cargo
in some way are limited to 60 minutes and the remaining ones to
10-15mins.

lucasmerlin committed:
Merge branch 'main' into demo-layout

justincredible committed:
Add missing fields from `Layout` to `LayoutSettings`

justincredible committed:
Merge branch 'demo-layout' into demo-layout2

justincredible committed:
Merge branch 'main' into demo-layout

justincredible committed:
Update snapshot

lucasmerlin reviewed (changes requested):
Nice! But we should add some note that explains that main_justify breaks when adding more than a single element and then only show a single widget when main_justify is enabled.

justincredible committed:
Allow user to control resize behaviour

justincredible committed:
Allocate area works too

justincredible committed:
Add warning label for main justify option

The warning disappears when main wrap is set, and when resize area is restricted unless main justify is set.

justincredible committed:
Allow single element in demo

justincredible committed:
Update snapshot

justincredible committed:
Apply clippy & fmt

justincredible committed:
Merge branch 'main' into demo-layout

justincredible committed:
Add missing case for warning label

The label will appear as long as the user is within one step of triggering the infinite auto-resize.

justincredible commented:
https://github.com/user-attachments/assets/28d17937-f32e-4374-add8-4e86428e4742
